scipy/interpolate/_aaa.py

```diff
@@ -0,0 +1,353 @@
+    --------
+    pade : Padé approximation
+
+    Notes
+    -----
+    At the :math:`m` th iteration, the rational approximation in the AAA algorithm takes
+    the barycentric form
+
+    .. math::
+
+        r(z) = \frac{n(z)}{d(z)} =
+        \frac{\sum_{j=1}^m\frac{w_jf_j}{z-z_j}}{\sum_{j=1}^m\frac{w_j}{z-z_j}},
+
+    where :math:`z_1,\dots,z_m` are real or complex support points selected from `z`,
+    :math:`f_1,\dots,f_m` are a set of real or complex data values, and
+    :math:`w_1,\dots,w_m` are real or complex weights. The algorithm then proceeds to
+    select the next support point :math:`z_{m+1}` is selected from the remaining
+    unselected points in `z` such that the nonlinear residual :math:`|f(z) - n(z)/d(z)|`
+    is maximised. The weights are selected to solve the least-squares problem
+
+    .. math::
+
+        \text{minimise}\lVert fd - n \rVert \quad \text{subject to} \quad
+        \sum_{i=1}^{m+1}w_i = 1,
+
+    over the unselected points in `z`.
+
+    References
+    ----------
+    .. [1] Y. Nakatsukasa, O. Sete, and L. N. Trefethen, "The AAA algorithm for
+            rational approximation", SIAM J. Sci. Comp. 40 (2018), A1494-A1522.
+
+    Examples
+    --------
+
+    Here we reproduce a number of the numerical examples from [1]_ as demonstration
+    of the functionality offered by this method.
+
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy.interpolate import AAA
+
+    For the first example we approximate the gamma function on ``[-3.5, 4.5]`` by
+    extrapolating from 100 samples in ``[-1.5, 1.5]``.
+
+    >>> from scipy.special import gamma
+    >>> sample_points = np.linspace(-1.5, 1.5, num=100)
+    >>> r = AAA(gamma, sample_points)
+    >>> z = np.linspace(-3.5, 4.5, num=1000)
+    >>> fig, ax = plt.subplots()
+    >>> ax.plot(z, gamma(z), label="Gamma")
+    >>> ax.plot(sample_points, gamma(sample_points), label="Sample points")
+    >>> ax.plot(z, r(z).real, '--', label="AAA approximation")
+    >>> ax.set(xlabel="z", ylabel="r(z)", ylim=[-8, 8], xlim=[-3.5, 4.5])
+    >>> ax.legend()
+    >>> plt.show()
+
+    We can also view the poles of the rational approximation and their residue:
+
+    >>> for i in np.argsort(r.poles):
+    ...     print(f"{r.poles[i]=:.3e}, {r.residues[i]=:.3e}")
+    r.poles[i]=-3.816e+00+0.000e+00j, r.residues[i]=3.658e-02+0.000e+00j
+    r.poles[i]=-3.003e+00+0.000e+00j, r.residues[i]=-1.692e-01-0.000e+00j
+    r.poles[i]=-2.000e+00+0.000e+00j, r.residues[i]=5.000e-01+0.000e+00j
+    r.poles[i]=-1.000e+00+0.000e+00j, r.residues[i]=-1.000e+00+0.000e+00j
+    r.poles[i]=5.858e-17+0.000e+00j, r.residues[i]=1.000e+00+0.000e+00j  # may vary
+    r.poles[i]=4.775e+00-3.069e+00j, r.residues[i]=-8.113e-01-2.302e+00j
+    r.poles[i]=4.775e+00+3.069e+00j, r.residues[i]=-8.113e-01+2.302e+00j
+    r.poles[i]=5.291e+00-9.737e-01j, r.residues[i]=8.733e-01+1.070e+01j
+    r.poles[i]=5.291e+00+9.737e-01j, r.residues[i]=8.733e-01-1.070e+01j
```

Comment:
consider just `>>> r.poles` and `>>> r.residues` --- this is both not very readable for a user and does not play well with smoke_docs.

Reply:
Good point, done